### ISSUE TITLE
java.lang.NoSuchMethodError: com.netflix.client.AbstractLoadBalancerAwareClient: method <init>()V not found

### ISSUE THREAD

**Issue by wololock** [spring-cloud/spring-cloud-starters#9]
Hello,

I'm using `spring-cloud-starter-eureka` to orchestrate the routing of a few microservices in the application I currently work on and one issue occurred when I moved from version `1.0.0.M3` to `1.0.0.RC1`. Here is the full stack trace - https://gist.github.com/wololock/a6c08ff38115c2804711

What I do is to call the load balancer to return the microservice's url:

    ServiceInstance instance = loadBalancerClient.choose('pdf-gen')
    String encodedUrl = URLEncoder.encode(pdfUrl, 'UTF-8')

It seems like the problem is caused by this call: https://github.com/spring-cloud/spring-cloud-netflix/blob/1.0.0.RC1/spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/ribbon/RibbonClientConfiguration.java#L66

Currently I just get back to `1.0.0.M3`. Is there anything I can do to help you resolve that issue? Or maybe it's something related to my app? I'll appreciate any suggestion. Thanks!

**Comment by spencergibb** [comment]
what version of `ribbon-httpclient` do you have on your classpath?  We use `2.0-RC13`?

**Comment by dsyer** [comment]
Looks like just a classpath issue. You have the wrong version of ribbon-httpclient on your classpath (RC1 and snapshots use 2.0-RC13). Can you post your POM? Or if you aren't using Maven try that.

**Comment by wololock** [comment]
Thanks for your suggestion, it was very helpful. I'm using Gradle, here are my dependencies - https://gist.github.com/wololock/ffc3428a7553a2040979
I figured out, that `com.netflix.ribbon:ribbon-httpclient:0.3.12` is currently in the classpath. I have also tracked the dependencies report, here is the result of the investigation:

`ribbon-eureka:2.0-RC13` uses `eureka-client:1.1.142` - https://github.com/Netflix/ribbon/blob/2.0-RC13/build.gradle#L89

`eureka-client:1.1.142` uses `ribbon-httpclient:0.3.12` - https://github.com/Netflix/eureka/blob/1.1.142/build.gradle#L7

The easiest workaround that works for me is to add

    compile "com.netflix.ribbon:ribbon-httpclient:2.0-RC13"

to the `build.gradle` and then this conflict is resolved by using `2.0-RC13`. The question is - why is it that sometimes it woks and sometimes it doesn't? :)
